src/idiomorph.js

```diff
@@ -646,10 +649,14 @@ var Idiomorph = (function () {
           ctx,
         );
       } else {
-        morphAttributes(oldNode, newContent, ctx);
-        if (!ignoreValueOfActiveElement(oldNode, ctx)) {
-          // @ts-ignore newContent can be a node here because .firstChild will be null
-          morphChildren(ctx, oldNode, newContent);
+        if (!ctx.keepInputValues || !oldNode.isEqualNode(newContent)) {
```

Comment:
I see that the `.isEqualNode` check is being run twice, once before morphing the attrs, and again afterwards before morphing the children. Why not just the first check? I tried removing the second check, and the test suite still passes. Is there a good reason to do it twice?

Reply:
Also, can we invert the conditional to `if (ctx.keepInputValues && oldNode.isEqualNode(newContent))`? I have a hard time mentally intuiting the `!conditional || !conditional`